Documentation/Enums/Enum Declarations.md

```diff
@@ -0,0 +1,161 @@
+# Enum Declarations
+
+## Feature Overview
+
+An enumeration, or enum for short, is a static listing of named values.
+Enums are a quick and efficient method of storing a limited range of possibilities as integers.
+
+
+## Commands
+
+### `enum end`
+
+`enum end`
+
+The `enum end` command ends declaring an enum type.
+
+### `enum start`
+
+`enum start` `:` `name`
+
+The `enum start` command starts declaring an enum type.
+
+### `enum value`
+
+`enum value` `:` `name`*`[, value]`*
+
+The `enum value` command declares a value of an enum inside its declaration.
+If provided, the value must be an integer.
+
+
+## Usage
+
+```
+enum start : Direction
+enum value : Unknown 0
+enum value : Horizontal
+enum value : Vertical
+enum end
+```
+
+### CSharp
+
+```
+enum Direction {
+    Unknown = 0;
+    Horizontal;
+    Vertical;
+}
+```
+
+### Java
+
+```
+enum Direction {
+    Unknown = 0;
+    Horizontal;
+    Vertical;
+}
+```
+
+### Python
+
+```
+enum Direction {
+    Unknown = 0;
+    Horizontal;
+    Vertical;
+}
+```
+
+### Ruby
+
+```
+enum Direction {
+    Unknown = 0;
```

Comment:
I'm not familiar with Ruby too much, but does this syntax work?


Possibly relevant: http://stackoverflow.com/questions/75759/enums-in-ruby